homeassistant/components/hunterdouglas_powerview/__init__.py

```diff
@@ -143,3 +124,31 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok:
         hass.data[DOMAIN].pop(entry.entry_id)
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate to new config version."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+    if config_entry.version == 1:
+        new_data = {**config_entry.data}
+
+        websession = async_get_clientsession(hass)
+        pv_request = AioRequest(
+            config_entry.data[CONF_HOST], loop=hass.loop, websession=websession
+        )
+
+        # fetch and add the API version
+        hub = Hub(pv_request)
+        await hub.query_firmware()
+        new_data[CONF_API_VERSION] = hub.api_version
+
+        config_entry.version = 2
+        hass.config_entries.async_update_entry(config_entry, data=new_data)
+
+        _LOGGER.info(
+            "Migration of %s from version 1 to version %s successful",
+            DOMAIN,
+            config_entry.version,
+        )
+
+    return True
```

Comment:
It looks like we can likely avoid migrating the version and just set the field with `async_update_entry`. We want to avoid migrating unless we are completely changing the whole structure of the integration as it makes downgrades very difficult for users.

Reply:
no worries - just tried to follow the doco

so we want to add something like this to ``__init__.py``  and leave ``config_entry.version = 1``?
```
def async_setup_entry(....)
....
    try:
        async with async_timeout.timeout(10):
            hub = Hub(pv_request)
            await hub.query_firmware()
            device_info = await async_get_device_info(hub)

            if CONF_API_VERSION not in config:
               config[CONF_API_VERSION] = hub.api_version
               hass.config_entries.async_update_entry(config_entry, data=config)
``

then update ``config_flow.py`` to add for new setups automatically ?

Reply:
That should work 👍